ts/components/FiscalCodeComponent.tsx

```diff
@@ -221,41 +322,91 @@ const styles = StyleSheet.create({
 });
 
 export default class FiscalCodeComponent extends React.Component<Props> {
-  private renderFrontContent(profile: UserProfile, isLandscape: boolean) {
+  private renderItem(
+    content: string,
+    fullStyle: StyleProp<ViewStyle>,
+    landscapeStyle: StyleProp<ViewStyle>,
+    isLandscape: boolean
+  ) {
+    return (
+      <Text
+        robotomono={true}
+        bold={true}
+        style={[
+          isLandscape
+            ? [styles.landscapeText, landscapeStyle]
+            : [styles.fullText, fullStyle]
+        ]}
+      >
+        {content.toUpperCase()}
+      </Text>
+    );
+  }
+
+  private renderFrontContent(
+    profile: UserProfile,
+    municipality: MunicipalityState,
+    isLandscape: boolean
+  ) {
+    const fiscalCodeData = extractFiscalCodeData(
+      profile.fiscal_code,
+      municipality
+    );
+
     return (
       <React.Fragment>
-        <Text
-          bold={true}
-          style={[
+        {this.renderItem(
+          profile.fiscal_code,
+          styles.fullFiscalCodeText,
+          styles.landscapeFiscalCodeText,
+          isLandscape
+        )}
+
+        {this.renderItem(
+          profile.family_name,
+          styles.fullLastNameText,
+          styles.landscapeLastNameText,
+          isLandscape
+        )}
+
+        {this.renderItem(
+          profile.name,
+          styles.fullNameText,
+          styles.landscapeNameText,
+          isLandscape
+        )}
+
+        {pot.isSome(municipality.data) &&
```

Comment:
I just realized right now we don't need this check since `fiscalCodeData.denominazione` and `fiscalCodeData.siglaProvincia` are string (not optional). 
If the pot is **_not some_** they will be set to empty string inside `extractFiscalCodeData` function.

For `fiscalCodeData.gender` we could pass `fiscalCodeData.gender || ""`
What do you think?

Reply:
I passed `municipality` to  `extractFiscalCodeData` to get data all data on the same object (`fiscalCodeData`) but I get municipality from the client request. 
I preferred to add these controls to avoid one or more empty text tags are rendered.